### PR TITLE
CGE-121_Implement_AdvertisementTitleDto

### PR DESCRIPTION
**Pull request by drw85**
- Created new class AdvertisementTitleDto;
- Created new endpoint '/adv/thumbnail' to retrieve such dtos;

### CODE REVIEW

File: src/main/java/com/hillel/items_exchange/dto/AdvertisementTitleDto.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import com.hillel.items_exchange.model.Location;
+import lombok.*;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@EqualsAndHashCode
```
Comment (rpkyrych) [inline src/main/java/com/hillel/items_exchange/dto/AdvertisementTitleDto.java:11]: You should exclude **advertisementId** from equals and hashcode

File: src/main/java/com/hillel/items_exchange/service/impl/AdvertisementServiceImpl.java [hunk 2]
```diff
@@ -149,6 +157,19 @@ public void setDefaultImage(Advertisement advertisement, Long imageId, User owne
         }.getType());
     }
 
+    private List<AdvertisementTitleDto> mapAdvertisementsToTitleDto(Collection<Advertisement> advertisements) {
+        return advertisements.stream().map(advertisement ->
+                AdvertisementTitleDto.builder()
```
Comment (rpkyrych) [inline src/main/java/com/hillel/items_exchange/service/impl/AdvertisementServiceImpl.java:162]: You should extract builder into another method **buildAdvertisementTitle** or something similar to and use is here with link to the method ::

Comment (rpkyrych) [reply src/main/java/com/hillel/items_exchange/service/impl/AdvertisementServiceImpl.java:162]: Or you could use **  BeanUtils.copyProperties** but you all the fields names should be the same

File: src/test/java/com/hillel/items_exchange/controller/AdvertisementControllerTest.java [hunk 3]
```diff
@@ -45,6 +45,16 @@ void findPaginated_shouldBeThrownValidationException() throws Exception {
         assertTrue(mvcResult.getResponse().getContentAsString().contains("must be greater than or equal to 0"));
     }
 
+    @Test
+    void findPaginatedAsThumbnails_shouldBeThrownValidationException() throws Exception {
```
Comment (rpkyrych) [inline src/test/java/com/hillel/items_exchange/controller/AdvertisementControllerTest.java:49]: We already have negative test for pagination. Would you mind to create positive test instead and validate the fields?

**Review by rpkyrych — CHANGES_REQUESTED** [review]
Looks great but few things needs to be changed.

File: src/main/java/com/hillel/items_exchange/dto/AdvertisementTitleDto.java (hunk 1 above)
Comment (drw85) [reply src/main/java/com/hillel/items_exchange/dto/AdvertisementTitleDto.java:11]: Yeah. You are right.

File: src/main/java/com/hillel/items_exchange/service/impl/AdvertisementServiceImpl.java (hunk 2 above)
Comment (drw85) [reply src/main/java/com/hillel/items_exchange/service/impl/AdvertisementServiceImpl.java:162]: Good point. Thnx.

File: src/test/java/com/hillel/items_exchange/controller/AdvertisementControllerTest.java (hunk 3 above)
Comment (drw85) [reply src/test/java/com/hillel/items_exchange/controller/AdvertisementControllerTest.java:49]: Agree. It must be done. I'll do it.

**Review by rpkyrych — APPROVED** [review]

**Review by Jack11M — APPROVED** [review]

**Comment by drw85** [comment]
- Created new class AdvertisementTitleDto;
- Created new endpoint '/adv/thumbnail' to retrieve such dtos;
- Covered with tests;

**Pull request merged**